src/Module/Contact.php

```diff
@@ -0,0 +1,1123 @@
+		}
+	}
+
+	private static function updateContactFromProbe($contact_id)
+	{
+		$contact = DBA::selectFirst('contact', ['uid', 'url', 'network'], ['id' => $contact_id, 'uid' => local_user()]);
+		if (!DBA::isResult($contact)) {
+			return;
+		}
+
+		$uid = $contact["uid"];
+
+		$data = Probe::uri($contact["url"], "", 0, false);
+
+		// "Feed" or "Unknown" is mostly a sign of communication problems
+		if ((in_array($data["network"], [Protocol::FEED, Protocol::PHANTOM])) && ($data["network"] != $contact["network"])) {
+			return;
+		}
+
+		$updatefields = ["name", "nick", "url", "addr", "batch", "notify", "poll", "request", "confirm",
+			"poco", "network", "alias"];
+		$update = [];
+
+		if ($data["network"] == Protocol::OSTATUS) {
+			$result = Model\Contact::createFromProbe($uid, $data["url"], false);
+
+			if ($result['success']) {
+				$update["subhub"] = true;
+			}
+		}
+
+		foreach ($updatefields AS $field) {
+			if (isset($data[$field]) && ($data[$field] != "")) {
+				$update[$field] = $data[$field];
+			}
+		}
+
+		$update["nurl"] = normalise_link($data["url"]);
+
+		$query = "";
+
+		if (isset($data["priority"]) && ($data["priority"] != 0)) {
+			$query = "'priority' => '" . intval($data["priority"]) . "'";
+		}
+
+		foreach ($update AS $key => $value) {
+			if ($query != "") {
+				$query .= ", ";
+			}
+
+			$query .= "'" . $key . "' => '" . DBA::escape($value) . "'";
+		}
+
+		if ($query == "") {
+			return;
+		}
+
+		$r = DBA::update('contact', $query, ['id' => $contact_id, 'uid' => local_user()]);
+
+		// Update the entry in the contact table
+		Model\Contact::updateAvatar($data['photo'], local_user(), $contact_id, true);
+
+		// Update the entry in the gcontact table
+		Model\GContact::updateFromProbe($data["url"]);
+	}
+
+	private static function blockContact($contact_id)
+	{
+		$blocked = !Model\Contact::isBlockedByUser($contact_id, local_user());
+		Model\Contact::setBlockedForUser($contact_id, local_user(), $blocked);
+	}
+
+	private static function ignoreContact($contact_id)
+	{
+		$ignored = !Model\Contact::isIgnoredByUser($contact_id, local_user());
+		Model\Contact::setIgnoredForUser($contact_id, local_user(), $ignored);
+	}
+
+	private static function archiveContact($contact_id, $orig_record)
+	{
+		$archived = (($orig_record['archive']) ? 0 : 1);
+		$r = DBA::update('contact', ['archive' => $archived], ['id' => $contact_id, 'uid' => local_user()]);
+
+		return DBA::isResult($r);
+	}
+
+	private static function dropContact($orig_record)
+	{
+		$a = get_app();
+
+		$r = q("SELECT `contact`.*, `user`.* FROM `contact` INNER JOIN `user` ON `contact`.`uid` = `user`.`uid`
+			WHERE `user`.`uid` = %d AND `contact`.`self` LIMIT 1",
+			intval($a->user['uid'])
+		);
+		if (!DBA::isResult($r)) {
+			return;
+		}
+
+		Model\Contact::terminateFriendship($r[0], $orig_record, true);
+		Model\Contact::remove($orig_record['id']);
+	}
+
+	public static function content($update = 0)
```

Comment:
Oh, that's important. On /mod/contacts this is really used. So I guess we have to postpone this a little bit. Especially, since there also is this "update_contacts.php" function as well. This is used for live updating the contacts posts.

Reply:
How is it used exactly?

Reply:
Then it's been taken care of here: 

https://github.com/friendica/friendica/blob/bfa05156ca5096fe9db8ca4a66ca32ef4cbf0ca5/mod/update_contacts.php#L16-L18

Reply:
We have to have a look into the javascript as well - and the update mustn't have this name, I guess.

Reply:
The Javascript can't call `contact_content()` directly with a parameter, it has to go through the router in `index.php` which doesn't provide any parameter beyond the App.

Reply:
Yes, but "contacts" is renamed to "contact" and the update still is named "update_contacts" - and in this JS there is some JS voodoo with "contacts", I think.

Reply:
I just checked, everything seems clear.

Reply:
Please have a look here: https://github.com/friendica/friendica/blob/develop/view/js/main.js#L393

Reply:
Yep, I saw it but we didn't touch it. It still outputs `<div id="live-contacts"></div>` which means the auto-update would still work.

Reply:
But isn't the page renamed to "contact" without an "s"?

Reply:
The page, yes, but the HMTL id is still `live-contacts` and the Javascript looks for `#live-contacts` and call `/update_contacts` (that still exists).

Reply:
Actually I found a wrong use statement in `mod/update_contacts.php` thanks to your prodding about the JS part. It has nothing to do directly with it but it is now fixed, thanks!